src/app.py

```diff
@@ -109,18 +113,41 @@ async def send_scene():
 
     await asyncio.gather(*future_sends)
 
-
 def close_processes():
     for process_data in state.processes.values():
         process_data.process.terminate()
 
+
+async def tick_trigger():
+    if time.time() - state.last_tick >= 1:
+        await tick()
+        state.last_tick = time.time()
+
+async def wait_for_next_tick():
+    next_tick_number = state.tick_number + 1
+    while state.tick_number != next_tick_number:
+        await asyncio.sleep(0.1)
+
+async def spawn_process_for_new_agent(object: Object):
```

Comment:
`-> NoReturn`

Reply:
Let's add `NoReturn` to functions in separate refactor PR